src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+        {
+            const int Messages = 1;
+            foreach (bool append in DataflowTestHelpers.BooleanValues)
+            {
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable);
+                var values = new int[Messages];
+                var targets = new ActionBlock<int>[Messages];
+                for (int i = 0; i < Messages; i++)
+                {
+                    int slot = i;
+                    targets[i] = new ActionBlock<int>(item => values[slot] = item);
+                    tb.LinkTo(targets[i], new DataflowLinkOptions { MaxMessages = 1, Append = append });
+                }
+
+                tb.PostRange(0, Messages);
+                tb.Complete();
+                await tb.Completion;
+
+                for (int i = 0; i < Messages; i++)
+                {
+                    Assert.Equal(
+                        expected: append ? i : Messages - i - 1,
+                        actual: values[i]);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestReceivesAsyncEnumerable()
+        {
+            for (int test = 0; test < 2; test++)
+            {
+                var tb = new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i * 2, 1));
+                tb.PostRange(0, 5);
+
+                for (int i = 0; i < 5; i++)
+                {
+                    Assert.Equal(expected: i * 2, actual: await tb.ReceiveAsync());
+                }
+
+                Assert.False(tb.TryReceive(out _));
+                Assert.False(tb.TryReceiveAll(out _));
+            }
+        }
+
+        [Fact]
+        public async Task TestCircularLinkingAsyncEnumerable()
+        {
+            const int Iters = 200;
+
+            var tcs = new TaskCompletionSource<bool>();
+            IAsyncEnumerable<int> body(int i)
+            {
+                if (i >= Iters) tcs.SetResult(true);
+                return AsyncEnumerable.Repeat(i + 1, 1);
+            }
+
+            TransformManyBlock<int, int> tb = new TransformManyBlock<int, int>(body);
+
+            using (tb.LinkTo(tb))
+            {
+                tb.Post(0);
+                await tcs.Task;
+                tb.Complete();
+            }
+        }
+
+        [Fact]
+        public async Task TestProducerConsumerAsyncEnumerable()
+        {
+            foreach (TaskScheduler scheduler in new[] { TaskScheduler.Default, new ConcurrentExclusiveSchedulerPair().ConcurrentScheduler })
+                foreach (int maxMessagesPerTask in new[] { DataflowBlockOptions.Unbounded, 1, 2 })
+                    foreach (int boundedCapacity in new[] { DataflowBlockOptions.Unbounded, 1, 2 })
+                        foreach (int dop in new[] { 1, 2 })
+                            foreach (int elementsPerItem in new[] { 1, 3, 5 })
+                            {
+                                const int Messages = 50;
```

Comment:
```suggestion
                                const int messages = 50;
```
private constants should be written in lower case